core/runtime/src/lib.rs

```diff
@@ -63,55 +110,34 @@ mod text;
 #[doc(inline)]
 pub use text::{TextDecoder, TextEncoder};
 
-pub mod url;
-
+#[cfg(feature = "fetch")]
+pub mod fetch;
 pub mod interval;
+pub mod url;
 
-/// Options used when registering all built-in objects and functions of the `WebAPI` runtime.
-#[derive(Debug)]
-pub struct RegisterOptions<L: Logger> {
-    console_logger: L,
-}
-
-impl Default for RegisterOptions<DefaultLogger> {
-    fn default() -> Self {
-        Self {
-            console_logger: DefaultLogger,
-        }
-    }
-}
-
-impl RegisterOptions<DefaultLogger> {
-    /// Create a new `RegisterOptions` with the default options.
-    #[must_use]
-    pub fn new() -> Self {
-        Self::default()
-    }
-}
+mod options;
 
-impl<L: Logger> RegisterOptions<L> {
-    /// Set the logger for the console object.
-    pub fn with_console_logger<L2: Logger>(self, logger: L2) -> RegisterOptions<L2> {
-        RegisterOptions::<L2> {
-            console_logger: logger,
-        }
-    }
-}
+pub use options::RegisterOptions;
 
 /// Register all the built-in objects and functions of the `WebAPI` runtime.
 ///
 /// # Errors
 /// This will error is any of the built-in objects or functions cannot be registered.
-pub fn register(
+pub fn register<#[cfg(feature = "fetch")] F: fetch::Fetcher, L: Logger + 'static>(
```

Comment:
I don't think it's a good pattern to change the generics of a function depending on the enabled features. I think it's much better to gate the implementors of `Fetched` themselves, then leave the `ErrorFetcher` as the default if the feature is disabled.

Reply:
For this I'll probably sync with you on Elements as I think I'm in the right here (though the actual code could likely improve).

What I want out of this feature is to remove ALL the code you don't need, which is a surprising amount. If the feature doesn't reduce the code compiled then we should just include fetch all the time.

The trait is dependent on `JsResponse`, `JsRequest`, `JsHeaders` etc. Basically all except the actual `fetch` function. That's a lot of code to include when the feature is disabled. So I'd like to remove all mentions of `fetch` and co from the API when the feature is not specified.